[Lee Kang Wei] iP

lkwlkww opened the pull request:
# MumBot
> "Homely." - A wise man

## Code snippet
```java
  public static void main(String[] args) throws DukeException {
      // Welcome message
      Ui.print("MumBot: Hi dear! You are precious <3\n");

      /**
       * Handling of the .txt file containing the list of tasks.
       */
      String filePath = "/Users/kw/Library/Mobile Documents/com~apple~CloudDocs/MODS/cs2103/projects/ip/data/duke.txt";
      Storage storage = new Storage(filePath);
      tasks = storage.getTasks();

      String input;
      Scanner sc = new Scanner(System.in);
      input = Ui.awaitUserInput(sc);

      while (true) {
          boolean notTerminated;

          try {
              notTerminated = Parser.settleInput(input, tasks);
          } catch (DukeException e) {
              Ui.print(e + "\n");
              input = Ui.awaitUserInput(sc);
              continue;
          }

          if (notTerminated) {
              input = Ui.awaitUserInput(sc);
          } else {
              break;
          }
      }

      storage.save();
      System.exit(0);
  }

```

## Commands
<i>Valid commands have to start with one of the following words</i>
- todo
- deadline
- event
- mark
- unmark
- delete
- list
- Bye
- find

## Instructions for usage
1. Build the .jar file using the source code [here](https://github.com/lkwlkww/ip)
2. Run the .jar file
3. Begin usage! 😄 
4. Enter `Bye` into the command prompt to end the MumBot programme.

## Completed features
- [x] Add find command
- [ ] Create GUI

lkwlkww committed:
ignore vim temp files

lkwlkww committed:
ignore .class files (java bytecode)

lkwlkww committed:
working version of Duke that satisfies Level-1

lkwlkww committed:
add note that all inputs are considered as Strings

lkwlkww committed:
finish up level-1 implementation

lkwlkww committed:
finish level-2 implementation

lkwlkww committed:
finish implementation for Level-3

lkwlkww committed:
build new class for commands; create package for commands and tasks

lkwlkww committed:
minor changes/bugfixes to Command subclasses

lkwlkww committed:
refactored Duke.java to work with the new tasks and commands packages

lkwlkww committed:
add text to be printed

lkwlkww committed:
delete old code + make existing code neater

lkwlkww committed:
bugfixes

lkwlkww committed:
bugfix

lkwlkww committed:
bugfixes, complete Level-4

lkwlkww committed:
added command to run javac on all package directories

lkwlkww committed:
create basic inputs for testing

lkwlkww committed:
add on to response for list command

lkwlkww committed:
update files for basic testing

lkwlkww committed:
Create DukeException class

lkwlkww committed:
implement error handling using DukeException

lkwlkww committed:
updated test files

lkwlkww committed:
minor change

lkwlkww committed:
add delete command basic functionalities

lkwlkww committed:
finished implementation of a delete command

lkwlkww committed:
update testing input and output

lkwlkww committed:
refactor Tasks into a single class containing a Type enum

lkwlkww committed:
change switch case syntax to fit coding standard

lkwlkww committed:
change import statements to fit coding standard

lkwlkww committed:
fix bug

lkwlkww committed:
finish implementation of saving and loading tasks

lkwlkww committed:
recognise dates in a specific format in deadline commands

lkwlkww committed:
Merge branch 'branch-Level-7'

lkwlkww committed:
Merge branch 'branch-Level-8'

lkwlkww committed:
change commands package name to command

lkwlkww committed:
change exceptions package name to exception

lkwlkww committed:
change tasks package name to task

lkwlkww committed:
update imported package names

lkwlkww committed:
create ui class and package

lkwlkww committed:
Add print method; Change method headers

lkwlkww committed:
add method to await user input; change logic of print methods

lkwlkww committed:
refactor all classes to use the Ui class

lkwlkww committed:
refactor all classes to use a new storage class for task storage operations

lkwlkww committed:
refactor all classes to use the new TaskList class

lkwlkww committed:
refactor classes to work with new Parser class

lkwlkww committed:
update files required for parse.Parser class

lkwlkww committed:
create duke package

lkwlkww committed:
add gradle support

lkwlkww committed:
Merge branch 'feature-creation'

lkwlkww committed:
add JUnit tests for TaskList and ByeCommand

lkwlkww committed:
change filepath for saved tasks to a hardcoded path

lkwlkww committed:
add class header for ui

lkwlkww committed:
update build.gradle

lkwlkww committed:
update method headers

lkwlkww committed:
add functionality for finding tasks using a search string

lkwlkww committed:
change indentation levels to fit coding standard

lkwlkww committed:
Merge branch 'branch-A-CodingStandard'

lkwlkww committed:
fix merge conflicts

lkwlkww committed:
Merge branch 'branch-A-JavaDoc' of github.com:lkwlkww/ip into branch-A-JavaDoc

lkwlkww committed:
Merge branch 'branch-A-JavaDoc'

zacchaeuschok commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+        input = Ui.awaitUserInput(sc);
+
+        while (true) {
+            boolean notTerminated;
```
Perhaps use a prefix such as `is` for the boolean variable.

lkwlkww replied:
Thanks! I missed this. I think changing the way its named would make it a bit more intuitive as well

zacchaeuschok commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+  @Override
+  public boolean performAction() {
+    TaskList foundTasks = tasks.searchUsingString(commandArgs[1]);
+    Ui.print("Sweetie, here is the list of tasks containing your search input <333");
```
Hey, I like how to took the effort to personalise your bot 👍

lkwlkww replied:
😃 thank you

zacchaeuschok commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+   * @return A <code>TaskList<Task></code> containing the tasks,
+   */
+  public TaskList searchUsingString(String searchString) {
+    ArrayList<Task> foundTasks = new ArrayList<Task>();
```
Perhaps can be written as `ArrayList<Task> foundTasks = new ArrayList<>();` because of type inference?

lkwlkww replied:
Actually I decided to write it this way to make the code more explicit, for other people who might read through the code.

I'm not too sure which is better - what are your thoughts?

zacchaeuschok commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,99 @@
+     */
+    public String updateStatus(boolean changeTo) {
+        if (isDone) {
+            if (changeTo) {
```
Perhaps you could check for `changeTo` being equal to `isDone` to reduce the number of conditions?

lkwlkww committed:
change `notTerminated` variable to a more intuitive name

lkwlkww committed:
resolve gradle checkstyleMain errors

lkwlkww committed:
alter build.gradle to include javafx dependencies

eesung00 commented on src/main/java/duke/parse/Parser.java:
```diff
@@ -0,0 +1,195 @@
+            Command command = new ListCommand(splitInput, tasks);
+            return command.performAction();
+
+        } else if (
+                action.equals("mark")
+                || action.equals("unmark")
```
Perhaps you can group mark, unmark and delete error together to shorten the method? Since they are checking the same conditions.

eesung00 commented on src/main/java/duke/parse/Parser.java:
```diff
@@ -0,0 +1,195 @@
+                action.equals("mark")
+                || action.equals("unmark")
+        ) {
+            // Throw an error if the formatting for the 'mark' or 'unmark' command is wrong
```
I like how you comment the function of a block of code in long methods. It improves the readability of your code!

lkwlkww committed:
create GUI for duke

lkwlkww committed:
change parseString method to use varargs

lkwlkww committed:
revert changes

lkwlkww committed:
Add java assert statement into duke.java

The current programme lacks java assert statements.

Java assert statements should be added to detect possible
bugs in the code.

Let's add a java assert statement into duke.java that ensures that
<code>tasks</code> is not null, after loading the saved tasks.

lkwlkww committed:
Merge pull request #2 from lkwlkww/branch-A-Assertions

Add java assert statement into duke.java

lkwlkww committed:
update build.gradle

lkwlkww committed:
refactor code to improve code quality

lkwlkww committed:
Refactors the code to improve code quality.

Code has certain defects in term of code quality.

Good code quality is needed to ensure understandability.

Let's refactor the code such that code quality is improved.

lkwlkww committed:
Merge pull request #3 from lkwlkww/branch-A-CodeQuality

Refactors the code to improve code quality.

lkwlkww committed:
Add functionality of assigning priority to tasks

lkwlkww committed:
Merge branch 'master' of github.com:lkwlkww/ip

lkwlkww committed:
fix bug for tasklist

lkwlkww committed:
Merge branch 'branch-C-Priority'

lkwlkww committed:
add ui.png

lkwlkww committed:
update message for wrong inputs

lkwlkww committed:
add userguide
